src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/domain/BlogPostRepository.java

```diff
@@ -16,7 +16,7 @@
 
     List<BlogPost> findByPublishedDateAfterAndApprovedTrueOrderByPublishedDateAsc(LocalDateTime publishedDate);
 
-    @Query("from BlogPost bp order by bp.publishedDate desc")
+    @Query("from BlogPost bp order by bp.approved desc, bp.publishedDate desc")
```

Comment:
Are you sure that it will put not moderated (approved = null) first and then all other by published date?